### PR TITLE
Use pillow for gif conversion

### PR DESCRIPTION
**Pull request by jamienoss**
Also:

 * Remove literal path delimiters
 * Don't remove dir containing individual frames

### CODE REVIEW

File: hcipy/plotting/animation.py [hunk 1]
```diff
@@ -1,43 +1,77 @@
...
+
+	def convert(self):
+		self.convert2gif(self.filename, self.path_to_frames, self.framerate, n_files2convert=self.n_frames)
+
 	def close(self):
-		if not self.closed:
-			command = ['convert', '-delay', str(int(100/self.framerate)), 
-						'-loop', '0', self.filename + '_frames/*.png', self.filename]
-			call(command)	
-			call(['rm', '-rf', self.filename + '_frames'])
```
Comment (jamienoss) [inline hcipy/plotting/animation.py]: I removed this but am happy to put back if you want it to remain.

We'll want to not have it do this but I can just inherit and override.

**Comment by jamienoss** [comment]
@ehpor Hey, I was just wondering how interested you were in this and if so, what your time frame for a next release would be?

I'm just wondering if I should just go ahead and merge https://github.com/spacetelescope/hicat-package/pull/91 (unless you spot issues).

**Comment by jamienoss** [comment]
@ehpor Sorry dude, I don't have time to add tests to address the coveralls fail.

**Comment by ehpor** [comment]
I've scheduled an hour to work on hcipy tomorrow, so I'll review it then. As for release schedule: we are working on a polarization upgrade for simplifying full Stokes propagation, so I wanted to release a new version then. This should take ~2-3 weeks still. We're still working on speed improvements and unit testing for that.

**Comment by jamienoss** [comment]
Ok cool, thank you.

File: hcipy/plotting/animation.py (hunk 1 above)
Comment (ehpor) [reply hcipy/plotting/animation.py]: Make it an option in `__init__()` called `cleanup` with default `True`.

File: hcipy/plotting/animation.py [hunk 2]
```diff
@@ -1,43 +1,83 @@
...
 		self.framerate = framerate
 
-		if not os.path.exists(filename + '_frames/'):
-			os.mkdir(filename + '_frames/')
-		self.frame_number = 0
-	
+		self.path_to_frames = self.filename + "_frames"
+		if not os.path.exists(self.path_to_frames):
+			os.mkdir(self.path_to_frames)
+		self.n_frames = 0
```
Comment (ehpor) [inline hcipy/plotting/animation.py]: Call it `num_frames`

File: hcipy/plotting/animation.py [hunk 3]
```diff
@@ -1,43 +1,83 @@
...
-		
-		dest = self.filename + '_frames/%05d.png' % self.frame_number
-		self.frame_number += 1
+
+		dest = os.path.join(self.path_to_frames, '%05d.png' % self.n_frames)
 
 		if arr is None:
 			if fig is None:
-				fig = plt.gcf()
+				fig = matplotlib.pyplot.plt.gcf()
```
Comment (ehpor) [inline hcipy/plotting/animation.py]: This won't work. Remove the plt.

File: hcipy/plotting/animation.py [hunk 4]
```diff
@@ -1,43 +1,83 @@
...
+		search_pattern = os.path.join(src_file_path, "*."+src_file_suffix)
+		files = glob.glob(search_pattern)
+		files.sort()
+
+		if n_files2convert is not None and len(files) != n_files2convert:
+			raise OSError("Expected {} files but found {}".format(n_files2convert, len(files)))
+
+		# Open all frames to convert
+		frames = []
+		for file in files:
```
Comment (ehpor) [inline hcipy/plotting/animation.py]: Don't use the Python keyword `file` as a variable name. Also for other occurrences.

File: hcipy/plotting/animation.py [hunk 5]
```diff
@@ -1,43 +1,83 @@
...
+			if cmap is not None:
 				arr = matplotlib.cm.get_cmap(cmap)(arr, bytes=True)
 			
 			imageio.imwrite(dest, arr, format='png')
-	
+
+		self.n_frames += 1
+
+	@staticmethod
+	def convert2gif(dest_filename, src_file_path, framerate,  src_file_suffix="png", n_files2convert=None):
```
Comment (ehpor) [inline hcipy/plotting/animation.py]: I don't like to abbreviate "to" as "2". Can you write this out? Ie. `convert_to_gif()`, etc...

File: hcipy/plotting/animation.py (hunk 3 above)
Comment (ehpor) [reply hcipy/plotting/animation.py]: Also, the same change needs to be made in the `add_frame()` of `FFMpegWriter` below to not break that function.

File: hcipy/plotting/animation.py [hunk 6]
```diff
@@ -1,43 +1,83 @@
...
-		if not os.path.exists(filename + '_frames/'):
-			os.mkdir(filename + '_frames/')
-		self.frame_number = 0
-	
+		self.path_to_frames = self.filename + "_frames"
+		if not os.path.exists(self.path_to_frames):
+			os.mkdir(self.path_to_frames)
+		self.n_frames = 0
+
+	def __del__(self):
```
Comment (ehpor) [inline hcipy/plotting/animation.py]: Can you add this function to FFMpegWriter below as well?

File: hcipy/plotting/animation.py (hunk 5 above)
Comment (ehpor) [reply hcipy/plotting/animation.py]: Also, change `n_files2convert` to `num_files_to_convert`.

**Comment by ehpor** [comment]
I'll add docstrings myself after merging.

**Comment by jamienoss** [comment]
@ehpor Thanks for the review. Changes made, back to you.

**Pull request merged**